openedx/features/courseware/tests.py

```diff
@@ -0,0 +1,203 @@
+"""Unit tests for courseware"""
+
+from django.urls import reverse
+
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+from openedx.features.courseware import constants
+from student.tests.factories import UserFactory, CourseEnrollmentFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+
+class FetchPrePostAssessmentMixin(ModuleStoreTestCase, APITestCase):
+    """Mixin for fetch pre/post assessments score tests"""
+
+    # Tell Django to clean out all databases, not just default
+    multi_db = True
+
+    def basic_setup(self):
+        """Set up a simple course for testing pre/post assessments score api."""
+        self.course = CourseFactory.create(display_name='test_course', number='100')
+        self.user = UserFactory.create(username='dummy', password='foo')
+        self.add_pre_post_assessment_grading_policy()
+
+    def get_url(self, chapter):
+        """Method to create the url"""
+        return reverse(
+            'courseware:pre_post_assessments_score',
+            kwargs={
+                'course_id': self.course.id,
+                'chapter_id': chapter.location.block_id,
+            }
+        )
+
+    def add_pre_post_assessment_grading_policy(self):
+        """Add grading policy to mark assessments pre or post for testing"""
+        grading_policy = {
+            "GRADER": [
+                {
+                    "type": constants.PRE_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.PRE_ASSESSMENT_FORMAT,
+                    "weight": 0.0
+                },
+                {
+                    "type": constants.POST_ASSESSMENT_FORMAT,
+                    "min_count": 1,
+                    "drop_count": 0,
+                    "short_label": constants.POST_ASSESSMENT_FORMAT,
+                    "weight": 0.25
+                }
+            ]
+        }
+
+        self.course.grading_policy = grading_policy
+        self.update_course(self.course, self.user.id)
+        self.course = self.store.get_course(self.course.id)
+
+
+class FetchPrePostAssessmentScoreTestCase(FetchPrePostAssessmentMixin):
+    """Check fetch pre post assessment scores properly"""
+
+    def setUp(self):
+        super(FetchPrePostAssessmentScoreTestCase, self).setUp()
+        self.basic_setup()
+
+    def test_unauthorized_request(self):
```

Comment:
Lets move this test to `FetchPrePostAssessmentScoreForUnEnrolledUserTestCase`. At the start of test you can call logout on client